Remove gotestcover dependency, add vendor directory and verify checks

dobsonj opened the pull request:
This PR resolves a few issues for CI downstream in OpenShift -- hopefully in a way that will work for upstream CI as well.
1) `gotestcover` is deprecated and that repo is archived by the owner. These days we should be using `go test` and avoid the extra dependency. This was causing CI to fail downstream, because installing `gotestcover` without running `go mod vendor` causes the vendor directory to be out of sync and `make test` fails as shown here:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/22515/rehearse-22515-pull-ci-openshift-ibm-vpc-block-csi-driver-master-unit/1445439206124949504
2) The vendor directory is normally committed to the repo. Or at least, that is the standard practice set by k8s that the other CSI drivers follow. Without that, we always have to run `go mod vendor` from CI after cloning the repo, which shouldn't really be necessary.
3) We need to verify the vendor directory is up-to-date as part of the pre-merge checks, so I introduced a new `make verify` target along with supporting scripts under `hack/`. These came from https://github.com/openshift/aws-ebs-csi-driver with minor modification. This `make verify` target should be able to replace `make deps lint fmt`, but I left the existing make targets in place since upstream CI appears to be using them today.
The changes of interest are mainly in `Makefile` and `hack/*`. Everything in `vendor/*` was generated automatically by `go mod vendor`.

gnufied commented:
/ok-to-test

arahamad commented:
@dobsonj , Thank you for the PR. Can you please share some details or may be update the readme for these

1- How we will update the dependencies in vendor folder is there any automatic command/script which we added in the Makefile
2- If there is any new dependencies needs to be added then what steps a developer need to take

dobsonj commented:
> 1- How we will update the dependencies in vendor folder is there any automatic command/script which we added in the Makefile
> 2- If there is any new dependencies needs to be added then what steps a developer need to take
@arahamad It's just a matter of running `go mod vendor` when making any changes to `go.mod`. I can add a note to the readme. This is also why I added `hack/verify-vendor.sh` so that CI can catch any PR that updates `go.mod` without properly updating the vendor directory.
